src/components/CallView/shared/VideoBackground.vue

```diff
@@ -142,10 +222,68 @@ export default {
 		}
 	},
 
+	beforeDestroy() {
+		this.isDestroyed = true
+	},
+
 	methods: {
 		// Calculate the background blur based on the height of the background element
 		setBlur({ width, height }) {
-			this.blur = this.$store.getters.getBlurFilter(width, height)
+			this.blur = this.$store.getters.getBlurRadius(width, height)
+		},
+
+		generateBlurredBackgroundImage() {
+			// Reset image source so the width and height are adjusted to
+			// the element rather than to the previous image being shown.
+			this.$refs.backgroundImage.src = ''
+
+			let width = this.$refs.backgroundImage.width
+			let height = this.$refs.backgroundImage.height
+
+			// Restore the current background so it is shown instead of an empty
+			// background while the new one is being generated.
+			this.$refs.backgroundImage.src = this.blurredBackgroundImage
+
+			const sourceAspectRatio = this.blurredBackgroundImageSource.width / this.blurredBackgroundImageSource.height
+			const canvasAspectRatio = width / height
+
+			if (canvasAspectRatio > sourceAspectRatio) {
+				height = width / sourceAspectRatio
+			} else if (canvasAspectRatio < sourceAspectRatio) {
+				width = height * sourceAspectRatio
+			}
+
+			const cacheId = this.backgroundImageUrl + '-' + width + '-' + height + '-' + this.backgroundBlur
```

Comment:
it is likely that someone can overflow the cache by resizing the window gradually

maybe need to use stepping for different width/height (like using divisions of 1024 or something)

in general we probably only ever need to catch 2 or 3 dimensions: the one in grid view, the one in stripe view and the one in speaker view

Reply:
You would really need to resize the window very slowly and for a while to overflow the cache. But it would be indeed possible.

Therefore I will limit the cache of each VideoBackground to the last ten calculated dimensions or so (as whether the sidebars are open or not also affect the size of the background and, thus, the blur).